packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -1627,6 +1627,70 @@ const tests = {
         },
       ],
     },
+    {
+      code: normalizeIndent`
+        function MyComponent(props) {
+          useEffect(() => {
+            if(props.foo?.bar) {
+              console.log(props.foo.bar);
+            }
+          }, [])
+        }
+      `,
+      errors: [
+        {
+          message:
+            "React Hook useEffect has a missing dependency: 'props.foo?.bar'. " +
+            'Either include it or remove the dependency array.',
+          suggestions: [
+            {
+              desc: 'Update the dependencies array to be: [props.foo?.bar]',
+              output: normalizeIndent`
+                function MyComponent(props) {
+                  useEffect(() => {
+                    if(props.foo?.bar) {
+                      console.log(props.foo.bar);
+                    }
+                  }, [props.foo?.bar])
+                }
+              `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: normalizeIndent`
+        function ParentImpliedOptional(props) {
+          useEffect(() => {
+            if(props.foo?.bar) {
+              console.log(props.foo.baz);
+            }
+          }, [])
+        }
+      `,
+      errors: [
+        {
+          message:
+            "React Hook useEffect has missing dependencies: 'props.foo?.bar' and 'props.foo?.baz'. " +
+            'Either include them or remove the dependency array.',
+          suggestions: [
+            {
+              desc: 'Update the dependencies array to be: [props.foo?.bar, props.foo?.baz]',
+              output: normalizeIndent`
+                function ParentImpliedOptional(props) {
+                  useEffect(() => {
+                    if(props.foo?.bar) {
+                      console.log(props.foo.baz);
+                    }
+                  }, [props.foo?.bar, props.foo?.baz])
```

Comment:
What would happen if the property chain is completely disjoint e.g. 

```suggestion
                    if(props.foo?.bar?.baz) {
                      console.log(props.box.value);
                    }
                  }, [/* ?? */])
```